giskard/core/suite.py

```diff
@@ -151,6 +152,18 @@ def to_wandb(self, run: Optional["wandb.wandb_sdk.wandb_run.Run"] = None) -> Non
                 "repository for further assistance: https://github.com/Giskard-AI/giskard."
             ) from e
         run.log({"Test suite results/Test-Suite Results": wandb.Table(columns=columns, data=data)})
+      
+    def to_junit(self):
+        """Convert the test suite result to JUnit XML format."""
+        testsuite = Element('testsuite')
+
+        for test in self.results:
+            testcase = SubElement(testsuite, 'testcase', {'name': test.metric_name})
```

Comment:
The `metric_name` is optional and does not meet the `name` specification:

- name: The label of the rule.

Reply:
What do you propose I add instead of metric_name for the name parameter? @kevinmessiaen

Reply:
How should I get time for a particular test?

Reply:
To replace metric name I suggest having something like: `{test_name} ({param_name}: {param_value}, ...)`

Time is not tracked, so either we can keep it empty for now or add a time tracking mechanism in `Suite.run` method.

Reply:
Just changed the to_junit function to have more information.

Sample output:
```<?xml version="1.0" ?>
<testsuites tests="2">
   <testsuite name="Test test1 (metric=0.9)">
      <testcase name="Accuracy" time="0.9">
         <detail name="passed" value="True"/>
         <detail name="props" value="{'prop1': 'value1'}"/>
         <detail name="metric_name" value="Accuracy"/>
         <detail name="metric" value="0.9"/>
         <detail name="missing_count" value="0"/>
         <detail name="missing_percent" value="0"/>
         <detail name="unexpected_count" value="0"/>
         <detail name="unexpected_percent" value="0"/>
         <detail name="unexpected_percent_total" value="0"/>
         <detail name="unexpected_percent_nonmissing" value="0"/>
         <detail name="partial_unexpected_index_list" value="[]"/>
         <detail name="unexpected_index_list" value="[]"/>
         <detail name="number_of_perturbed_rows" value="0"/>
         <detail name="actual_slices_size" value="[]"/>
         <detail name="reference_slices_size" value="[]"/>
         <detail name="output_df" value="   A  B
0  1  4
1  2  5
2  3  6"/>
         <detail name="message" value="Message 1"/>
      </testcase>
   </testsuite>
   <testsuite name="Test test2 (metric=0.8)">
      <testcase name="Precision" time="0.8">
         <failure message="Test failed with metric of 0.8" type="TestFailed">
            <detail name="passed" value="False"/>
            <detail name="props" value="{'prop2': 'value2'}"/>
            <detail name="metric_name" value="Precision"/>
            <detail name="metric" value="0.8"/>
            <detail name="missing_count" value="0"/>
            <detail name="missing_percent" value="0"/>
            <detail name="unexpected_count" value="0"/>
            <detail name="unexpected_percent" value="0"/>
            <detail name="unexpected_percent_total" value="0"/>
            <detail name="unexpected_percent_nonmissing" value="0"/>
            <detail name="partial_unexpected_index_list" value="[]"/>
            <detail name="unexpected_index_list" value="[]"/>
            <detail name="number_of_perturbed_rows" value="0"/>
            <detail name="actual_slices_size" value="[]"/>
            <detail name="reference_slices_size" value="[]"/>
            <detail name="output_df" value="   A  B
0  1  4
1  2  5
2  3  6"/>
            <detail name="message" value="Message 2"/>
         </failure>
      </testcase>
   </testsuite>
</testsuites>```